model/model.go

```diff
@@ -748,9 +748,10 @@ type PartitionInfo struct {
 
 // GetNameByID gets the partition name by ID.
 func (pi *PartitionInfo) GetNameByID(id int64) string {
-	for _, def := range pi.Definitions {
-		if id == def.ID {
-			return def.Name.L
+	definitions := pi.Definitions
+	for i := range definitions {
```

Comment:
heh. please add a comment linking to this PR to explain why we need to loop like this.

```suggestion
	// do not convert this loop to `for _, def := range definitions`.
	// see https://github.com/pingcap/parser/pull/1072 for the benchmark.
	for i := range definitions {
```